tests/PHPStan/Reflection/Constant/RuntimeConstantReflectionTest.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Constant;
+
+use PhpParser\Node\Name;
+use PHPStan\Testing\PHPStanTestCase;
+use PHPStan\TrinaryLogic;
+
+class RuntimeConstantReflectionTest extends PHPStanTestCase
+{
+
+	public function dataDeprecatedConstants(): array
+	{
+		return [
+			[
+				new Name('\DeprecatedConst\FINE'),
+				TrinaryLogic::createNo(),
+				null,
+			],
+			[
+				new Name('\DeprecatedConst\MY_CONST'),
+				TrinaryLogic::createYes(),
+				null,
+			],
+			[
+				new Name('\DeprecatedConst\MY_CONST2'),
+				TrinaryLogic::createYes(),
+				"don't use it!",
+			],
+		];
+	}
+
+	/**
+	 * @dataProvider dataDeprecatedConstants
+	 */
+	public function testDeprecatedConstants(Name $constName, TrinaryLogic $isDeprecated, ?string $deprecationMessage): void
```

Comment:
I had no dataprovider before and refactored it to what it is now, as it felt repetitive.

With the current structure one could add a new const definition into the test-file and add expectations for this file in the dataprovider

Reply:
Ah, I got it. I'm so used to using yield in those that the array nesting confused me I think ans I bleived it would execute the test only once. Makes sense.